providers/implementations/signature/rsa.c

```diff
@@ -177,6 +177,23 @@ static int rsa_check_padding(int mdnid, int padding)
     return 1;
 }
 
+static int rsa_check_parameters(EVP_MD *md, PROV_RSA_CTX *prsactx)
+{
+    if (prsactx->pad_mode == RSA_PKCS1_PSS_PADDING) {
+        int max_saltlen;
+
+        /* See if minimum salt length exceeds maximum possible */
+        max_saltlen = RSA_size(prsactx->rsa) - EVP_MD_size(md);
+        if ((RSA_bits(prsactx->rsa) & 0x7) == 1)
```

Comment:
So if there were 9 bits it subtracts 1, but for 10..15 bits is doesnt?

Reply:
Frankly, I have *no* idea why this is done, I copied this from here:

https://github.com/openssl/openssl/blob/e307e616f25a6b7b0f343fc1e62a35b2cba888f3/crypto/rsa/rsa_pmeth.c#L829-L836

So if you tell me this is wrong, I'll take your word for it, but in that case, we should look at the rsa_pmeth.c code as well.

Reply:
I have no idea if it is correct.. Guess we will assume it is ok fo rnow.